pkg/volume/util/resize_util_test.go

```diff
@@ -0,0 +1,165 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type conditionMergeTestCase struct {
+	description    string
+	pvc            *v1.PersistentVolumeClaim
+	newConditions  []v1.PersistentVolumeClaimCondition
+	finalCondtions []v1.PersistentVolumeClaimCondition
+}
+
+func TestMergeResizeCondition(t *testing.T) {
+	currentTime := metav1.Now()
+
+	pvc := getPVC([]v1.PersistentVolumeClaimCondition{
+		{
+			Type:               v1.PersistentVolumeClaimResizing,
+			Status:             v1.ConditionTrue,
+			LastTransitionTime: currentTime,
+		},
+	})
+
+	noConditionPVC := getPVC([]v1.PersistentVolumeClaimCondition{})
+
+	conditionFalseTime := metav1.Now()
+
+	testCases := []conditionMergeTestCase{
+		{
+			description:    "when removing all conditions",
+			pvc:            pvc.DeepCopy(),
+			newConditions:  []v1.PersistentVolumeClaimCondition{},
+			finalCondtions: []v1.PersistentVolumeClaimCondition{},
+		},
+		{
+			description: "adding new condition",
+			pvc:         pvc.DeepCopy(),
+			newConditions: []v1.PersistentVolumeClaimCondition{
+				{
+					Type:   v1.PersistentVolumeClaimFileSystemResizePending,
+					Status: v1.ConditionTrue,
+				},
+			},
+			finalCondtions: []v1.PersistentVolumeClaimCondition{
+				{
+					Type:   v1.PersistentVolumeClaimFileSystemResizePending,
+					Status: v1.ConditionTrue,
+				},
+			},
+		},
+		{
+			description: "adding same condition with new timestamp",
+			pvc:         pvc.DeepCopy(),
+			newConditions: []v1.PersistentVolumeClaimCondition{
+				{
+					Type:               v1.PersistentVolumeClaimResizing,
+					Status:             v1.ConditionTrue,
+					LastTransitionTime: metav1.Now(),
```

Comment:
add time to this so it is distinctly different than currentTime

Reply:
fixed